azext_iot/common/utility.py

```diff
@@ -146,6 +150,44 @@ def process_json_arg(content: str, argument_name: str = "content", preserve_orde
         )
 
 
+def process_yaml_arg(path: str) -> dict:
+    """Primary processor of yaml file input"""
+
+    if not os.path.exists(path):
+        raise FileOperationError(
+            f"YAML file not found - Please ensure the path '{path}' is correct."
+        )
+
+    try:
+        import yaml
+
+        with open(path, "rb") as f:
+            return yaml.load(f, Loader=yaml.SafeLoader)
+    except Exception as ex:
+        raise InvalidArgumentValueError(
+            "Failed to parse yaml for file '{}' with exception:\n{}".format(path, ex)
+        )
+
+
+def process_toml_content(path: str) -> Dict[str, Any]:
```

Comment:
nit, do you think ending in `_content` is appropriate? Just curious since we have process_yaml_**arg** and process_json_**arg**.

Reply:
I can change it to `_arg` - I think I chose this initially since it was a bit different than JSON (which can be passed in as a string or a path) - though YAML is processed the same way so I just missed one of these 🙃